### PR TITLE
cyber_logger: fix a off-by-one problem

### PR DESCRIPTION
**Pull request by lx18233184051**

### CODE REVIEW

**Comment by lx18233184051** [comment]
if the link size is sizeof(path), then it could be OOW

File: cyber/common/global_data.cc [hunk 1]
```diff
@@ -41,7 +41,7 @@ namespace {
 const std::string& kEmptyString = "";
 std::string program_path() {
   char path[PATH_MAX];
-  auto len = readlink("/proc/self/exe", path, sizeof(path));
+  auto len = readlink("/proc/self/exe", path, sizeof(path) - 1);
```
Comment (storypku) [inline cyber/common/global_data.cc:44]: Good catch.

**Review by storypku — APPROVED** [review]

**Comment by storypku** [comment]
> if the link size is sizeof(path), then it could be OOW
BTW, what does oow mean ?

File: cyber/common/global_data.cc (hunk 1 above)
Comment (xiaoxq) [reply cyber/common/global_data.cc:44]: How about `boost::filesystem::canonical()`? But it will bind us to boost.

**Comment by xiaoxq** [comment]
> > if the link size is sizeof(path), then it could be OOW
> 
> BTW, what does oow mean ?

OOM maybe? I think it's more like stack overflow.

**Pull request merged**